### ISSUE TITLE
Support custom drivers

### ISSUE THREAD

**Issue by crcn** [Automattic/mongoose#3129]
It'd be nice if mongoose had an API for specifying drivers instead of setting `process.env. MONGOOSE_DRIVER_PATH`.  I'd be happy to make a PR for this feature if it seems useful.

**Comment by vkarpov15** [comment]
Yeah right now the drivers API is not very clearly defined and used in a slightly hacky way to support limited isomorphism. Would be great to support that in a more reasonable way, PRs welcome :)

**Comment by vkarpov15** [comment]
We now have a `mongoose.setDriver()` function and an example of a [production lib that uses a driver to talk to a different database](https://npmjs.com/package/@datastax/astra-mongoose), as well as [playground.mongoosejs.io](https://playground.mongoosejs.io/) which is powered by an in-browser driver.
